packages/babel-traverse/src/scope/lib/renamer.js

```diff
@@ -97,6 +116,30 @@ export default class Renamer {
     );
   }
 
+  parameterDefaultHasOuterBinding(path, state) {
+    const { oldName, scope } = state;
+    const params = path.get("params");
+    for (let i = 0; i < params.length; i++) {
+      const param = params[i];
+      if (param.isAssignmentPattern()) {
+        const right = param.get("right");
+        if (!state.paramDefaultOuterBinding) {
+          if (
+            right.isIdentifier() &&
+            right.node.body.name === oldName &&
```

Comment:
```suggestion
            right.node.name === oldName &&
```

Identifiers don't have `body`, right?
Also, this should have been caught by a test.

Reply:
Ahh, good catch. Fixed and testcase is added.